### PR TITLE
Enhance CreateEffectWarhead

### LINKED ISSUE #10629: CreateEffectWarhead: Support for multiple ImpactSound entries
ImpactSound currently only supports one value. If more are specified it won't play any impact sounds at all.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30314060-createeffectwarhead-support-for-multiple-impactsound-entries?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
This adds some randomness to the audio and visual effects of `CreateEffectWarhead` as requested by @DoGyAUT.

Also closes #10629.

### CODE REVIEW

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs [hunk 1]
```diff
@@ -17,17 +17,17 @@ namespace OpenRA.Mods.Common.Warheads
 {
 	public class CreateEffectWarhead : Warhead
 	{
-		[Desc("Explosion effect to use.")]
-		public readonly string Explosion = null;
+		[Desc("List of explosion effects that can be used.")]
+		public readonly List<string> Explosion = new List<string>();
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: This now needs to be `Explosions`.

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs [hunk 2]
```diff
@@ -17,17 +17,17 @@ namespace OpenRA.Mods.Common.Warheads
...
+		[Desc("Palette to use for explosion effect."), PaletteReference("UsePlayerPalette")]
+		public readonly string ExplosionPalette = "effect";
 
 		[Desc("Remap explosion effect to player color, if art supports it.")]
 		public readonly bool UsePlayerPalette = false;
 
-		[Desc("Sound to play on impact.")]
-		public readonly string ImpactSound = null;
+		[Desc("List of sounds that can be played on impact.")]
+		public readonly List<string> ImpactSound = new List<string>();
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: Likewise `ImpactSounds`.

Comment (reaperrr) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: And don't forget the upgrade rules.

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs (hunk 1 above)
Comment (pchote) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: Lets make these `string[]`s too, please.

**Comment by penev92** [comment]
Fixed `--check-yaml`, rebased and applied requested changes.
Also added another commit to clean up some of the Fort Lonestar and Drop Zone W weapon definitions.

**Comment by RoosterDragon** [comment]
:+1: Looks good to me.

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/63626aadca490f8425cfaf74be010f391239265f...757e6d5a04465261a48c5129a964e837b33114eb)

**Pull request merged**